lib/libzfs/libzfs_dataset.c

```diff
@@ -1231,22 +1231,6 @@ zfs_valid_proplist(libzfs_handle_t *hdl, zfs_type_t type, nvlist_t *nvl,
 		}
 
 		case ZFS_PROP_SPECIAL_SMALL_BLOCKS:
-			if (zpool_hdl != NULL) {
-				char state[64] = "";
-
-				/*
-				 * Issue a warning but do not fail so that
-				 * tests for setable properties succeed.
-				 */
-				if (zpool_prop_get_feature(zpool_hdl,
```

Comment:
I'm not understanding why this check needs to be removed?  Setting the `special_small_blocks` property on a dataset when no special vdevs are present serves no practical purpose.  Is there a use case that would benefit from omitting this warning?

Reply:
yep, i want all my backups on any system to have the same setting even if it makes no sense. if i later add a special vdev, these settings will take effect.

Reply:
See #8962
`zfs send -p` on a system with special classes, being received on a system without, will hard fail the transfer. I would call this a bug.

Reply:
@kpande @DeHackEd thanks for the `zfs send` context.  I'm still not understanding how this error message would cause a failure?  Is something expecting stderr to be empty?

Reply:
I actually noticed something in the original issue and followed up with my own testing. I can't reproduce an actual error/failure, and stderr shouldn't cause problems.